drivers/node/vsphere/vsphere.go

```diff
@@ -862,3 +862,336 @@ func (v *vsphere) RemoveNonRootDisks(n node.Node) error {
 
 	return nil
 }
+
+// StorageVmotion relocates the largest disks of a VM from one datastore to another within the same prefix group
+// With moveAllDisks true we will be moving all disks attached to a VM onto same Datastore
+// If moveAllDisks is set to False then we will choose the largest sized disk and move that only to a new Datastore
+func (v *vsphere) StorageVmotion(ctx context.Context, node node.Node, portworxNamespace string, moveAllDisks bool) error {
```

Comment:
We can discuss this. As of now this will be running in a controlled environment where we know the prefix and have set up datastores in that way accordingly. This could be taken up as future enhancement.